### DISCUSSION TITLE
Question: Is there any option to bring a specific index into view in a ListView?

### DISCUSSION THREAD

**Discussion by w-ahmad** [microsoft/microsoft-ui-xaml#9803] [Q&A]
First asked at community call chat but couldn't get answer or workaround. https://github.com/microsoft/microsoft-ui-xaml/discussions/9653#discussioncomment-9592584

When dealing with a `ListView` containing duplicate items in its `ItemsSource`, there isn't a direct method to specifically bring a duplicate item into view. For instance, if an item appears at index 30 and again at index 100, the challenge is how to scroll to the item at index 100. The commonly known method `ListViewBase.ScrollIntoView(object item)` will default to the item at index 30, not the one at index 100.

Manually calculating offset to a specific index is not possible when the items are of variable in size because all the containers are not initialized at once.

Test App: https://github.com/w-ahmad/ListViewScrollTest

**Comment by castorix** [comment]
If I do this, it scrolls to index 100 :

```
   var item = lv1.Items[100];
   lv1.ScrollIntoView(item);
   lv1.SelectedItem = item;
```
![ListView_ScrollIntoView](https://github.com/microsoft/microsoft-ui-xaml/assets/22345506/6cb260c5-86e1-4933-b3ec-2354eb5c47de)

**Reply by w-ahmad** [reply]
In the example the items are not duplicate I guess, the problem happens only when there are duplicate items in the ItemsSource. Here is a test application that demonstrates the problem: https://githu6b.com/w-ahmad/ListViewScrollTest

**Reply by castorix** [reply]
> In the example the items are not duplicate I guess, the problem happens only when there are duplicate items in the ItemsSource. Here is a test application that demonstrates the problem: https://githu6b.com/w-ahmad/ListViewScrollTest

Right.
But to select the right item, I added a "hidden" index in the ObservableCollection of the ItemsSource (a field not displayed in the ListView)

**Reply by w-ahmad** [reply]
Adding a hidden index to the data item is one method, and there could be other ways to address it. As I am facing this in a control distributed via the NuGet Package, I cannot dictate what users will set as the ItemsSource. The control is [WinUI.TableView](https://github.com/w-ahmad/WinUI.TableView), derived from ListView, and to let user select cells, I must scroll rows according to the selected cell.

**Comment by w-ahmad** [comment]
Would be very grateful if someone could suggest a solution or workaround.

**Comment by HHaoWang** [comment]
The problem of adding index numbers to list items view is very classic and still lacks a good solution after all these years. This need exists not only for ListView but also for various other collection controls. When I developed native Android apps, I was surprised by the implementation of RecyclerView, although its MVVM isn't as powerful as XAML, it's surprisingly simple to implement sequential numbering. Why not consider implementing this feature in first-party solutions?

**Comment by w-ahmad** [comment]
I found the internal implementation of `ScrollIntoView` method that is part of `Selector` class. 
https://github.com/microsoft/microsoft-ui-xaml/blob/66a7b0ae71c19f89c6a7d86a1986794ad1a1bf09/src/dxaml/xcp/dxaml/lib/Selector_Partial.cpp#L3103
The method accepts item object as the parameter, but it fetches the index of the given item from `Items` to pass to the next method.
https://github.com/microsoft/microsoft-ui-xaml/blob/66a7b0ae71c19f89c6a7d86a1986794ad1a1bf09/src/dxaml/xcp/dxaml/lib/Selector_Partial.cpp#L3152
The next method takes the index of item and do the stuff to bring it into view. It seems feasible to introduce a method in the `ListViewBase` class that would accept the item's index rather than the item itself.
@ranjeshj

**Reply by w-ahmad** [reply]
added it to the feature request #9890
